### PR TITLE
fix VariantSupportSuite; tests were not being run

### PR DESCRIPTION
**Pull request by ryan-williams**
iterator.size was being called before iterating over an iterator to check its elements; after size() the iterator is always empty, so the subsequent checks were never being run.

The "expected" data in several test cases was wrong, and I couldn't figure out what they might have originally been intended to point at in the BAM used in this file. I picked loci that exhibited similar behavior to what the test cases seemed to be intending to test, from nearby loci where possible, and fixed the cases up.

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/guacamole/412)

### CODE REVIEW

**Comment by timodonnell** [comment]
Cool, thanks for debugging this @ryan-williams !

LGTM modulo use of arrow characters

File: src/test/scala/org/hammerlab/guacamole/commands/VariantSupportSuite.scala [hunk 1]
```diff
@@ -14,99 +15,111 @@ class VariantSupportSuite extends GuacFunSuite with Matchers with TableDrivenPro
...
+  def nonDuplicateGatkReads(loci: String) =
+    TestUtil.loadReads(
+      sc,
+      "gatk_mini_bundle_extract.bam",
+      Read.InputFilters(
+        mapped = true,
+        nonDuplicate = true,
+        overlapsLoci = Some(LociSet.parse(loci))
+      ),
+      reference = grch37Reference
```
Comment (arahuja) [inline src/test/scala/org/hammerlab/guacamole/commands/VariantSupportSuite.scala]: I thin this will break now on merge?

**Comment by arahuja** [comment]
Great catch!

File: src/test/scala/org/hammerlab/guacamole/commands/VariantSupportSuite.scala (hunk 1 above)
Comment (ryan-williams) [inline src/test/scala/org/hammerlab/guacamole/commands/VariantSupportSuite.scala]: #401 didn't remove this `reference`; #415 does but it's not merged yet, so I'll leave it here and resolve conflicts as they arise

**Pull request merged**